hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaControllerManagement.java

```diff
@@ -386,6 +385,26 @@ public Target findOrRegisterTargetIfItDoesNotExist(final String controllerId, fi
                 .orElseGet(() -> createTarget(controllerId, address));
     }
 
+    @Override
+    @Transactional(isolation = Isolation.READ_COMMITTED)
+    @Retryable(include = ConcurrencyFailureException.class, exclude = EntityAlreadyExistsException.class, maxAttempts = Constants.TX_RT_MAX, backoff = @Backoff(delay = Constants.TX_RT_DELAY))
+    public Target findOrRegisterTargetIfItDoesNotExist(final String controllerId, final URI address,
```

Comment:
You could avoid code duplication in the findOrRegisterTargetIfItDoesNotExist methods by calling `return findOrRegisterTargetIfItDoesNotExist(controllerId, address, controllerId);`

Reply:
That's true, but I would not use the method like this as it's not "clear" why to handover the controllerId twice (for sure i know the implementation so i know why, but other developers don't). Are you fine with just keeping it?